pipelines/rbsa/tasks/zipcode_correlation.py

```diff
@@ -0,0 +1,120 @@
+import logging
+import pandas as pd
+import numpy as np
+from generics import task as t
+from generics.file_type_enum import SupportedFileReadType
+
+
+logger = logging.getLogger('LCTK_APPLICATION_LOGGER')
+
+
+class ZipcodeCorrelation(t.Task):
+    def __init__(self, name):
+        super().__init__(self)
+        self.name = name
+
+        self.input_artifact_full_zipcodes = 'full_zipcodes.csv'
+        self.input_artifact_projection_locations = 'PROJECTION_LOCATIONS.json'
+
+        # these will be used to generate list of input files
+        self.pre_data_files = [ 
+            { 'name': self.input_artifact_full_zipcodes, 'read_type': SupportedFileReadType.DATA },
+            { 'name': self.input_artifact_projection_locations, 'read_type': SupportedFileReadType.CONFIG }
+        ]
+
+        self.task_function = self._task
+        self.output_artifact_correlation_matrix = 'correlation_matrix.csv'
+        
+        self.data_map = None
+        self.df = None
+
+    def _save_data(self):
+        self.save_data(self.df)
+
+    def _get_data(self):
+        self.pre_data_map = self.load_data(self.pre_data_files) 
+        self.full_zipcodes = list(self.pre_data_map['full_zipcodes.csv']['zipcodes'])  
+        self.projection_locations = list(self.pre_data_map['PROJECTION_LOCATIONS.json']['cities'].keys())
+
+        self.data_files = []
+
+        for location in self.full_zipcodes:
+            self.data_files.append({ 'name': 'tmy_base/'+str(location)+'.csv', 'read_type': SupportedFileReadType.DATA })
+
+        for location in self.projection_locations:
+            self.data_files.append({ 'name': 'tmy_target/'+str(location)+'.csv', 'read_type': SupportedFileReadType.DATA })
+
+        self.data_map = self.load_data(self.data_files)       
+
+    def _task(self):
+        self._get_data()
+        logger.info(self.df)
+
+        correlation_metrics = ['Temperature', 'Solar Zenith Angle', 'GHI', 'DHI', 'DNI', 'Wind Speed', 'Wind Direction', 'Relative Humidity']    
+        correlation_matrix = pd.DataFrame(0, index=self.projection_locations, columns=self.full_zipcodes)
+
+        for base in self.full_zipcodes:
+            for target in self.projection_locations:
+
+                base_filename = 'tmy_base/'+str(base)+'.csv'
+                target_filename = 'tmy_target/'+str(target)+'.csv'
```

Comment:
use f'' strings for these instead of concatenating them

Reply:
fixed